File: Coding/Java/Set1/FirstRepeatingElement.java
package Set1;

import java.util.HashSet;

public class FirstRepeatingElement {

    public static int firstRepeating(int[] arr){
        HashSet<Integer> val = new HashSet<>();
        for(int num : arr){
            if(val.contains(num)){
                return num;
            }
            val.add(num);
        }
        return -1;
    }

    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 3, 3, 6, 6};
        int result = firstRepeating(arr);
        if(result!=-1)
            System.out.println(result);
        else
            System.out.println("There is no repeating element");
    }
}

File: Coding/Python/Set1/FirstRepeatingElement.py
def first_repeating(arr):
    temp_set = set()
    for val in arr:
        if val in temp_set:
            return val
        temp_set.add(val)

print(first_repeating([1, 2, 3, 3, 3, 6, 6]))
print(first_repeating([1, 6, 2, 3, 3, 3, 6, 6]))

